metaflow/plugins/datatools/s3/s3.py

```diff
@@ -819,6 +819,75 @@ def _head():
 
         return list(starmap(S3Object, _head()))
 
+    def delete(self, key):
+        """
+        Delete a single object in S3.
+        Parameters
+        ----------
+        key : str
+            Object to delete. It can be an S3 url or a path suffix.
+        """
+
+        url = self._url(key)
+        src = urlparse(url)
+
+        def _delete(s3, tmp):
+            if not src.path.lstrip("/"):
+                raise TypeError("'None' type object cannot be deleted.")
+            else:
+                return s3.delete_object(Bucket=src.netloc, Key=src.path.lstrip("/"))
+
+        self._one_boto_op(_delete, url, create_tmp_file=False)
+
+    def delete_many(self, keys):
+        """
+        Delete multiple objects from s3
+        Parameters
+        ----------
+        keys: List[str]
+            List of keys to delete.
+        """
+
+        urls = [self._url(key) for key in keys]
+        src = [urlparse(u) for u in urls]
+
+        Bucket_names = set([s.netloc for s in src])
+        if len(Bucket_names) > 1:
+            raise Exception("Cannot delete from multiple buckets at once")
+        else:
+            bucket_name = Bucket_names.pop()
+
+            resp = self._s3_client.delete_objects(
+                Bucket=bucket_name,
+                Delete={
+                    "Objects": [{"Key": s.path.lstrip("/")} for s in src],
+                    'Quiet': True
+                }
+            )
+
+        return resp
+
+    def delete_recursive(self, key):
+        """
+        Delete multiple objects given a prefix
+        Parameters
+        ----------
+        key: Prefix to delete
+        """
+
+        url = self._url(key)
+        src = urlparse(url)
+
+        def _delete(s3, tmp):
+            bucket = s3.Bucket(src.netloc)
+            if not src.path.lstrip('/'):
+                raise TypeError("'None' type object cannot be deleted.")
+            else:
+                for key_ in bucket.list(prefix=src.path.lstrip("/")):
```

Comment:
For this one though, I think it may be preferable to go the s3op route (ie: use a bunch of subprocesses) to parallelize it a bit. We could also potentially do a hybrid approach where we use s3op to get the listing using s3op (like in `list_recursive`) and then fallback on the `delete_many` implementation above where we delete all files in one fell swoop (modulo any restrictions on that as mentioned earlier).